File: UI/student-affairs/src/app/models/student.model.ts
import { StudentClass } from "./student-class.model";
import { StudentSubject } from "./student-subject.model";
import { Subject } from "./subject.model";

export interface Student {
    student_ID: number;
    studentClass_ID: number;
    studentClass_Name: string;
    student_Name: string;
    student_Address: string;
    student_BirthDate: Date;
    student_EmailAddress: string;
    classesList:StudentClass[];
    studentClass:StudentClass;
    studentSubjectsList:StudentSubject[];
    subjectsList:StudentSubject[];
  }

File: Students.Model/CustomModels/StudentModel.cs
using Students.Model.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Students.Model.CustomModels
{
    public class StudentModel
    {
        public int Student_ID { get; set; }
        public int StudentClass_ID { get; set; }
        public string StudentClass_Name { get; set; }
        public string Student_Name { get; set; }
        public string Student_Address { get; set; }
        public DateTime Student_BirthDate { get; set; }
        public string Student_EmailAddress { get; set; }
        public List<StudentClassModel> ClassesList { get; set; }
        public List<StudentSubjectModel> StudentSubjectsList { get; set; }
    }
}
